tasks/install.yml

```diff
@@ -0,0 +1,388 @@
+
+- name: Add nodejs repositoriy
+  apt_repository:
+    repo: "deb https://deb.nodesource.com/node_8.x artful main"
+
+- name: Update apt repositories cache
+  apt:
+    update_cache: yes
+  changed_when: false
+  when: ansible_pkg_mgr == "apt"
+
+- name: Install required packages
+  apt:
+    name: "{{item}}"
+  with_items:
+    - sudo
+    - build-essential
+    - fakeroot
+    - dpkg-dev
+    - zlib1g-dev
+    - libyaml-dev
+    - libssl-dev
+    - libgdbm-dev
+    - libre2-dev
+    - libreadline-dev
+    - libncurses5-dev
+    - libffi-dev
+    - curl
+    - openssh-server
+    - checkinstall
+    - libxml2-dev
+    - libxslt-dev
+    - libcurl4-openssl-dev
+    - libicu-dev
+    - logrotate
+    - python-docutils
+    - pkg-config
+    - cmake
+    - software-properties-common
+    - ruby2.4
+    - ruby2.4-dev
+    - nodejs
+    - golang-go
+    - yarn
+    - python-psycopg2
+    - git
+    - libpq-dev
+    - libpq5
+  when: ansible_pkg_mgr == "apt"
+
+- name: Add gitlab user
+  user:
+      name: git
+      shell: /bin/nologin
+      comment: 'GitLab'
+      state: present
+      groups: redis
+
+- name: Install bundler gem
+  gem:
+    name: bundler
+  become: yes
+  become_user: git
+
+- name: "Create database user {{gitlab_db_user}}"
+  postgresql_user:
+    login_host: "{{postgres_run_directory}}"
+    name: "{{gitlab_db_user}}"
+    role_attr_flags: CREATEDB
+
+- name: "Create GitLab database: {{gitlab_db_name}}"
+  postgresql_db:
+      login_host: "{{postgres_run_directory}}"
+      name: "{{gitlab_db_name|mandatory}}"
+      owner: "{{gitlab_db_user|mandatory}}"
+      login_user: "{{gitlab_db_user}}"
+
+- name: "Create pg_tgrm extension"
+  postgresql_ext:
+      login_host: "{{postgres_run_directory}}"
+      name: pg_trgm
+      db: "{{gitlab_db_name}}"
+
+- name: Creates redis socket directory
+  file:
+    path: "{{redis_unixsocket | dirname }}"
+    state: directory
+    owner: redis
+    group: redis
+    mode: 0755
+
+- name: Clone gitlab to {{gitlab_install_path | mandatory}}
+  git:
+    depth: 1
+    repo: https://gitlab.com/gitlab-org/gitlab-ee.git
+    dest: "{{ gitlab_install_path | mandatory }}"
+    version: "{{ gitlab_version_number | mandatory }}"
+    force: yes
+  when: gitlab_version == "ee"
+
+- name: Clone gitlab to {{gitlab_install_path}}
+  git:
+    depth: 1
+    repo: https://gitlab.com/gitlab-org/gitlab-ce.git
+    dest: "{{ gitlab_install_path }}"
+    version: "{{ gitlab_version_number }}"
+    force: yes
+  when: gitlab_version == "ce"
+
+- name: Set owner of gitlab directory
+  file:
+    path: "{{ gitlab_install_path }}"
+    owner: git
+    recurse: yes
+
+- name: Place gitlab config
+  template: 
+    src: templates/gitlab.yml.j2
+    dest: "{{ gitlab_install_path }}/config/gitlab.yml"
+
+- name: Copy secrets file
+  template:
+    src: templates/secrets.yml.j2
+    dest: "{{ gitlab_install_path }}/config/secrets.yml"
+    mode: 0600
+    owner: git
+
+- name: Set owner of log directory
```

Comment:
Can't you own the directories recursively?

Reply:
i dont want to own all subfolders